Implemented core java assignment

ztturner opened the pull request:
Completed implementing the questions required for the assignment.

ztturner committed:
Implemented acronym, triangle, scrabble, armstrong, ISBN, pangram

Implemented making acronyms, determining type of triangle, determining scrabble score, determining armstrong numbers, validating ISBN-10, determining pangrams.

ztturner committed:
Implemented word count, binary search, rotational cipher, Atbash cipher, sum of multiples, and Luhn

Implemented counting words for a string, the binary search algorithm, the rotational cipher to rotate strings,
the Atbash cipher with both encoding and decoding, finding the sum of multiples, and Luhn validation

ztturner committed:
Implemented phone number, pig latin, prime factors, finding primes, and solving word problems

Implemented the cleaning of phone number, translating to pig latin, finding primes and prime factors, and solving word problems.
Also made a small change to fix the Luhn validation

ztturner committed:
Separated rotational cipher generation and added comments

Separated the code that generates the map for the rotational cipher into its own method.
Added a private variable in the RotationalCipher class that holds the map for the pre-rotated and rotated characters.
This variable is populated when the constructor is called.
Also added some comments to the top of the custom methods.

Quinn-Donnelly reviewed (changes requested):
If you could please remove the tests related to the Time question. Travis won't like those unimplemented tests once we merge. Other than that looks good.

ztturner committed:
Removed time test cases

Removed test cases relating to the time question and actually saved the method header comments for the additional methods.
